src/bin/ion/commands/stats.rs

```diff
@@ -0,0 +1,189 @@
+use crate::commands::{CommandIo, IonCliCommand, WithIonCliArgument};
+use anyhow::Result;
+use clap::{ArgMatches, Command};
+use ion_rs::*;
+use ion_rs::{AnyEncoding, IonInput, IonType, SystemReader, SystemStreamItem};
+use lowcharts::plot;
+use std::cmp::max;
+
+pub struct StatsCommand;
+
+impl IonCliCommand for StatsCommand {
+    fn is_porcelain(&self) -> bool {
+        false
+    }
+
+    fn name(&self) -> &'static str {
+        "stats"
+    }
+
+    fn is_stable(&self) -> bool {
+        false
+    }
+
+    fn about(&self) -> &'static str {
+        "Print the analysis report of the input data stream, including the total number of top-level values, their minimum, maximum, and mean sizes,\n\
+        and plot the size distribution of the input stream. The report should also include the number of symbol tables in the input stream,\n\
+        the total number of different symbols that occurred in the input stream, and the maximum depth of the input data stream.\n\
+        Currently, this subcommand only supports data analysis on binary Ion data."
+    }
+
+    fn configure_args(&self, command: Command) -> Command {
+        command.with_input()
+    }
+    fn run(&self, _command_path: &mut Vec<String>, args: &ArgMatches) -> Result<()> {
+        CommandIo::new(args).for_each_input(|_output, input| {
+            let mut reader = SystemReader::new(AnyEncoding, input.into_source());
+            analyze(&mut reader, &mut std::io::stdout())
+        })
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+struct Output {
+    size_vec: Vec<f64>,
+    symtab_count: i32,
+    symbols_count: usize,
+    max_depth: usize,
+}
+
+fn analyze<Input: IonInput>(
+    reader: &mut SystemReader<AnyEncoding, Input>,
+    mut writer: impl std::io::Write,
+) -> Result<()> {
+    let out = analyze_data_stream(reader);
+    // Plot a histogram of the above vector, with 4 buckets and a precision
+    // chosen by library. The number of buckets could be changed as needed.
+    let options = plot::HistogramOptions {
+        intervals: 4,
+        ..Default::default()
+    };
+    let histogram = plot::Histogram::new(&out.size_vec, options);
+    writeln!(
+        writer,
+        "The 'samples' field represents the total number of top-level value of input data stream. The unit of min, max ,avg size is bytes.\n\
```

Comment:
```suggestion
        "The 'samples' field represents the total number of top-level value of input data stream. The unit of min, max, avg size is bytes.\n\
```

Reply:
Rather than explain what 'samples' means here, let's just rename 'samples' to "top-level values".

Reply:
Sounds good, I will update in the next revision.

Reply:
`samples` cannot be renamed because all the statistical results are calculated and written in `stats` when we initialize [Histogram](https://docs.rs/lowcharts/latest/src/lowcharts/plot/histogram.rs.html#58). [Here](https://docs.rs/lowcharts/latest/src/lowcharts/stats/mod.rs.html#65) is the implementation of displaying `Stats`.  According to the implementation, `samples` is written while formatting the results of `stats`.

Reply:
I see. We could always format the Histogram ourselves, but this is probably fine for now.